examples/java/src/main/java/org/apache/beam/examples/common/WriteOneFilePerWindow.java

```diff
@@ -62,32 +64,38 @@ public PDone expand(PCollection<String> input) {
    */
   public static class PerWindowFiles extends FilenamePolicy {
 
-    private final String output;
+    private final String filenamePrefix;
 
-    public PerWindowFiles(String output) {
-      this.output = output;
+    public PerWindowFiles(String filenamePrefix) {
+      String filePrefixOnly;
+      try {
```

Comment:
Comment what's going on here, and especially why on an exception choosing "" is the right thing to do?

Also, consider moving this code out of the constructor and instead into the caller.